packages/vite/src/node/server/moduleGraph.ts

```diff
@@ -197,4 +224,86 @@ export class ModuleGraph {
     }
     return [url, resolvedId]
   }
+
+  async loadCache(): Promise<void> {
+    const start = isDebug ? Date.now() : 0
+    const cacheLocation = this.getCacheLocation()
+    if (!cacheLocation) {
+      isDebug && debugModuleGraph('Cache disabled, loadCache skipped')
+      return
+    }
+    if (!fs.existsSync(cacheLocation)) {
+      isDebug && debugModuleGraph('Cache not found')
+      return
+    }
+    const cache: CacheFormat = JSON.parse(
+      fs.readFileSync(cacheLocation, { encoding: 'utf-8' })
+    )
+    for (const [url, value] of Object.entries(cache)) {
+      const id = (await this.container.resolveId(url))?.id || url
+      let loadResult = await this.container.load(id)
+      if (!loadResult) {
+        try {
+          loadResult = await fs.promises.readFile(id, 'utf-8')
```

Comment:
seems like this loop can be improved by making that work parallelly (up to x amount of tasks at a time). as js file system interaction s pretty slow.
something like this is the opposite extreme:
```
await Promise.all(Object.entries(cache).map(async ([url, value]) => {
      const id = (await this.container.resolveId(url))?.id || url
      let loadResult = await this.container.load(id)
     ....
}))
```